create_shards.c

```diff
@@ -97,6 +106,35 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 		ereport(ERROR, (errmsg("unsupported partition method: %c", partitionMethod)));
 	}
 
+	partitionColumnVar = ColumnNameToColumn(distributedTableId, partitionColumnName);
+	partitionColumnTypeId = partitionColumnVar->vartype;
+	partitionColumnOpClassId = GetDefaultOpClass(partitionColumnTypeId, BTREE_AM_OID);
+
+	/* if there is no operator class for the partition column, error out */
+	if (partitionColumnOpClassId == InvalidOid)
+	{
+		ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
+						errmsg("cannot distribute relation: \"%s\"", tableName),
+						errdetail("Column \"%s\" is not eligible to be the distribution "
+								  "column.", partitionColumnName),
+						errhint("Try another column as the distribution column.")));
+	}
+
+	lessEqualOperatorId = GetOperatorByType(partitionColumnTypeId, BTREE_AM_OID,
+											BTLessEqualStrategyNumber);
+	greaterEqualOperatorId =  GetOperatorByType(partitionColumnTypeId, BTREE_AM_OID,
+												BTGreaterEqualStrategyNumber);
+
+	/* if partition column does not support <= and >= operators, error out*/
+	if (lessEqualOperatorId == InvalidOid ||  greaterEqualOperatorId == InvalidOid)
```

Comment:
I believe this check only makes sense for tables that use range partitioning: for hash partitioning, the shards are always defined using 32-bit integers, and we can always find the inequality operator for that.

If using range partitioning, shard boundaries _will_ be defined using the partition column type, so it makes sense in that case to check for inequality operators.

Can you change this section to do something like:

```
if using hash partitioning
  check for presence of hash operator for partition column type
else if using range partitioning
  check for existence of inequality operators for partition column type
else
  error out that partition type is unexpected
```